### PR TITLE
Add AddressBalance() to FFCAPI and a new REST API route to call it

### PR DESCRIPTION
**Pull request by matthew1001**
The aim of this PR is to add an address balance API call to the FFCAPI interface. This makes it possible for users of of the FFCAPI (such as FireFly Transaction Manager) to query and optionally surface via their own APIs.

Under this PR I have introduced a new REST API `/gastoken/balances/{address}` which calls `AddressBalance()` for the the configured blockchain connector. The use of `gastoken` in the API path is intended to be clear that it is unrelated to token balances.

Signed-off-by: Matthew Whitehead <matthew.whitehead@kaleido.io>

### CODE REVIEW

File: internal/tmmsgs/en_api_descriptions.go [hunk 1]
```diff
@@ -51,6 +51,7 @@ var (
 	APIEndpointPostEventStreamListenerReset = ffm("api.endpoints.post.eventstream.listener.reset", "Reset an event stream listener, to redeliver all events since the specified block")
 	APIEndpointPatchEventStreamListener     = ffm("api.endpoints.patch.eventstream.listener", "Update event stream listener")
 	APIEndpointDeleteEventStreamListener    = ffm("api.endpoints.delete.eventstream.listener", "Delete event stream listener")
+	APIEndpointGetAddressBalance            = ffm("api.endpoints.get.address.balance", "Get balance for a signer address")
```
Comment (peterbroadhurst) [inline internal/tmmsgs/en_api_descriptions.go:54]:
```suggestion
	APIEndpointGetAddressBalance            = ffm("api.endpoints.get.address.balance", "Get gas token balance for a signer address")
```

File: internal/tmmsgs/en_api_descriptions.go [hunk 2]
```diff
@@ -60,4 +61,5 @@ var (
 	APIParamTXSigner      = ffm("api.params.txSigner", "Return only transactions for a specific signing address, in reverse nonce order")
 	APIParamTXPending     = ffm("api.params.txPending", "Return only pending transactions, in reverse submission sequence (a 'sequenceId' is assigned to each transaction to determine its sequence")
 	APIParamSortDirection = ffm("api.params.sortDirection", "Sort direction: 'asc'/'ascending' or 'desc'/'descending'")
+	APIParamSignerAddress = ffm("api.params.signerAddress", "A signing address, for example to get the balance for")
```
Comment (peterbroadhurst) [inline internal/tmmsgs/en_api_descriptions.go:64]:
```
	APIParamSignerAddress = ffm("api.params.signerAddress", "A signing address, for example to get the gas token balance for")
```

File: pkg/fftm/route_get_address_balance.go [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+var getAddressBalance = func(m *manager) *ffapi.Route {
+	return &ffapi.Route{
+		Name:   "getBalance",
+		Path:   "/gastoken/balances/{address}",
+		Method: http.MethodGet,
+		PathParams: []*ffapi.PathParam{
+			{Name: "address", Description: tmmsgs.APIParamSignerAddress},
+		},
+		QueryParams:     nil,
```
Comment (peterbroadhurst) [inline pkg/fftm/route_get_address_balance.go:35]: Should there not be a query param for `blocktag` here?

**Review by peterbroadhurst — COMMENTED** [review]
Two minor suggestions just to clarify which balance is being queried (e.g. it's not a coin, but the underlying native gas token of the chain), and a question on whether the `blockTag` field that's added to the FFCAPI should be included on the REST API.

File: internal/tmmsgs/en_api_descriptions.go (hunk 2 above)
Comment (matthew1001) [reply internal/tmmsgs/en_api_descriptions.go:64]: Changed.

File: pkg/fftm/route_get_address_balance.go [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+var getAddressBalance = func(m *manager) *ffapi.Route {
+	return &ffapi.Route{
+		Name:   "getBalance",
+		Path:   "/gastoken/balances/{address}",
+		Method: http.MethodGet,
+		PathParams: []*ffapi.PathParam{
+			{Name: "address", Description: tmmsgs.APIParamSignerAddress},
+		},
+		QueryParams: []*ffapi.QueryParam{
+			{Name: "blockTag", Description: tmmsgs.APIParamBlockTag},
```
Comment (peterbroadhurst) [inline pkg/fftm/route_get_address_balance.go:36]: Tiny point, but while these are case insensitive we've previously tried to make the docs the lower-case value.

**Review by peterbroadhurst — APPROVED** [review]
One single-character change suggestion with approval

**Pull request merged**